recipe/meta.yaml

```
@@ -43,6 +43,7 @@ requirements:
     - geos
     - ghostscript
     - libgdal-jp2openjpeg
+    - libgdal-hdf5
```

Sort h-i-j. Can you also add a test for reading a NetCDF/HDF5 file?

```suggestion
    - libgdal-hdf5
    - libgdal-jp2openjpeg
```

---

> Can you also add a test for reading a NetCDF/HDF5 file?

I added a test in ff8d995968563736b1bc021cce8c1760b8be615e, and it worked without the `libgdal-hdf5` library, but if I add `libgdal-hdf5` in https://github.com/GenericMappingTools/pygmt/pull/3338/commits/bb0efe386680acc8231c70edfc1f9661a92a50f3, tests passed (except one failure which is not directly related to the issue).

I think the reason is, when reading a netCDF file, GMT calls the netcdf library, but in https://github.com/GenericMappingTools/pygmt/pull/3338, we're actually trying to read a netCDF file as an image (into a GMT_IMAGE container), which instead calls GDAL to read the file.

---

Right... so should we actually add `libgdal-hdf5` as a dependency here then? To be honest, that `test_clib_read_data_image_actual_grid` in https://github.com/GenericMappingTools/pygmt/pull/3338 is a bit of a wrird test since we're trying to read a 1-band grid as a GMT image.

---

> To be honest, that `test_clib_read_data_image_actual_grid` test in [GenericMappingTools/pygmt#3338](https://github.com/GenericMappingTools/pygmt/pull/3338) is a bit weird since we're trying to read a 1-band grid as a GMT image.

The original idea comes from https://github.com/GenericMappingTools/pygmt/pull/3318#issue-2394677303 to solve the issue https://github.com/GenericMappingTools/pygmt/pull/3115#discussion_r1667671439.

For any given file, we can call `Session.read_data` to read the file header only and determine the raster kind (`grid`, `image` or `cube`). It turns out, we can read ".nc" file as a GMT image and read ".tif" file as a GMT grid. So that test was added to understand what's exactly happening.

> so should we actually add libgdal-hdf5 as a dependency here then?

We still need `libgdal-hdf5`, because reading other image formats via GDAL will likely need the HDF5 library.

---

Or actually, can you add the test back and use the `=gd` modifier to force GMT to read the NetCDF file using GDAL? https://docs.generic-mapping-tools.org/6.5/reference/features.html#modifiers-to-read-and-write-grids-and-images-via-gdal

---

```
gmt grdcut @earth_relief_01d=gd -R-10/-9/3/5 -Greliefcut.nc
```
Test fails without libgdal-hdf5. See
https://github.com/conda-forge/gmt-feedstock/pull/293/checks?check_run_id=27695109707

---

Awesome, I'll apply this to the `dev` branch for the next dev version update next week or so.